src/machine.cpp

```diff
@@ -6034,11 +6066,14 @@ GcJclass* getDeclaringClass(Thread* t, GcClass* c)
 
 GcCallSite* resolveDynamic(Thread* t, GcInvocation* invocation)
 {
+  // This support for invokedynamic is limited to support for the LambdaMetafactory, as the method dispatch logic
+  // assumes the first three args are always as expected by that class. It doesn't do full indy support.
```

Comment:
Actually I realised that the comment is wrong and removed it - the other commit adding it back is some merge issue. The support is correct and general (well, not fully complete, but not entirely lambda specific). I'll get it fixed.